Automattic/woocommerce-payments
# Use wp.data for the evidence component

Fixes #903

## Changes proposed in this Pull Request

Refactor the Evidence component in such a way that it uses `wp.data` for the data layer instead of component-local state. The method to access and manage the data is also updated slightly to match some other parts of the codebase.

Took some inspiration from the Settings data layer, specifically the actions in `data/settings/actions.js` on how to manage the actions.


## Testing instructions

Since this is a refactor there should be no functional changes. To test the feature follow the critical path testing instructions for disputes:

- [Dispute created notifications](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#dispute-created-notifications)
- [Save evidence](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#save-evidence)
- [Winning dispute](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#winning-dispute)
- [Losing dispute](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#losing-dispute)

Make sure all the critical flows work correctly. Once that's done we need to verify some of the main functionality in the evidence component.


### "Discard changes" notification when leaving the evidence screen after edits

1. Go to **Payments > Disputes**, select a dispute that needs response, and click **Challenge dispute**.
2. Make changes to 1 or more of the evidence fields; try to remember the changes you make.
3. Click **Payments > Disputes** to try to navigation back to the disputes list.
4. You should see a pop-up asking you to confirm that you want to discard the changes you made.
5. Click **Cancel**. You should still be on the evidence screen (i.e. you should not be taken to the disputes list.
6. Click **Payments > Disputes** again, make sure you see the same pop-up asking you to confirm that you want to discard changes.
7. Click **Accept**.
8. Make sure you're taken to the disputes list.
9. Go back to the same dispute, click **Challenge dispute**, and make sure the changes you made in Step 2 are no longer present.


### Saving evidence after changes shouldn't cause a notification that you're about to discard changes

1. Go to **Payments > Disputes**, select a dispute that needs response, and click **Challenge dispute**.
2. Make changes to 1 or more of the evidence fields; try to remember the changes you make.
3. Scroll down to the bottom of the page and click **Save evidence**.
4. You should be taken back to the disputes list without seeing a notification that you're about to discard changes.


### Submitting evidence after changes shouldn't cause a notification that you're about to discard changes

1. Go to **Payments > Disputes**, select a dispute that needs response, and click **Challenge dispute**.
2. Make changes to 1 or more of the evidence fields; try to remember the changes you make.
3. Scroll down to the bottom of the page and click **Submit evidence**.
4. You should see a notification asking you to confirm that you want to submit the evidence.
5. Click **Accept**.
4. You should be taken back to the disputes list without seeing a notification that you're about to discard changes.


### Accepting a dispute directly works

1. Go to **Payments > Disputes**, select a dispute that needs response, and click **Accept dispute**.
2. The dispute should be accepted (lost) without any errors.


-------------------


**Post merge**
